### PR TITLE
Move shim for Fedora to uefi only packages

### PR DESCRIPTION
**Pull request by DaanDeMeyer**

### CODE REVIEW

**Comment by lysliu** [comment]
@DaanDeMeyer I test it, new problems with this PR
```
Error: Transaction failed
Error: call to /usr/sbin/ldconfig failed.nError: call to /usr/sbin/iconvconfig failed.nError: call to /usr/sbin/ldconfig failed.n‣ "env HOME=/ dnf --assumeyes --best --releasever=39 --installroot=/var/tmp/mkosi-workspace8yszfeuj/root --setopt=keepcache=1 --setopt=logdir=/var/log --setopt=cachedir=/var/cache/dnf --setopt=persistdir=/var/lib/dnf --setopt=install_weak_deps=0 --setopt=check_config_file_age=0 '--disableplugin=*' --enableplugin=builddep --setopt=metadata_expire=never --forcearch=s390x --config=/etc/dnf/dnf.conf --setopt=reposdir=/etc/yum.repos.d --setopt=varsdir=/etc/dnf/vars install amd-ucode-firmware apt archlinux-keyring attr autoconf automake bash btrfs-progs bubblewrap ca-certificates ca-certificates centos-packager coreutils cpio curl-minimal debian-keyring distribution-gpg-keys dnf dnf-plugins-core dnf5 dnf5-plugins dosfstools e2fsprogs erofs-utils fedora-packager fedora-packager-kerberos fedora-review gcc gdb gettext git iproute iputils kernel-core less libtool make mock mock-centos-sig-configs mock-core-configs mtools nano openssh-clients openssh-server openssl pacman perf pkgconf python3-cryptography qemu-kvm-core rpm-build rpminspect rpminspect-data-centos rpminspect-data-fedora shadow-utils socat squashfs-tools strace strace swtpm systemd systemd-container systemd-networkd systemd-resolved systemd-udev systemd-ukify tar tmux util-linux virtiofsd xfsprogs xz zstd zypper" returned non-zero exit code 1.
root:~/mkosi-main$ uname -a
Linux boosters1.fyre.ibm.com 5.15.0-92-generic #102-Ubuntu SMP Wed Jan 10 09:33:48 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
root:~/mkosi-main$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

**Comment by DaanDeMeyer** [comment]
@lysliu Have you installed and enabled qemu user mode emulation for s390x? That's required to build s390x images on other architectures.

**Comment by lysliu** [comment]
> @lysliu Is this on an s390x host system?

I have both x86 and s390x host system,  I'm happy to verify/test any scenario needed with your help.

**Comment by lysliu** [comment]
> qemu user mode emulation

Sorry, I am a new learner about mkosi, could you help to provide a guide?
Or I will switch the work on the `s390x` host to verify this PR?

**Comment by DaanDeMeyer** [comment]
@lysliu Install the qemu-user package

**Comment by lysliu** [comment]
This is the fail message on `s390x` host
```
==> Updating trust database...
gpg: Note: third-party key signatures using the SHA1 algorithm are rejected
gpg: (use option "--allow-weak-key-signatures" to override)
gpg: marginals needed: 3  completes needed: 1  trust model: pgp
gpg: depth: 0  valid:   1  signed:   5  trust: 0-, 0q, 0n, 0m, 0f, 1u
gpg: depth: 1  valid:   5  signed:  98  trust: 0-, 0q, 0n, 5m, 0f, 0u
gpg: depth: 2  valid:  74  signed:  22  trust: 74-, 0q, 0n, 0m, 0f, 0u
gpg: next trustdb check due at 2024-04-10
Unknown command verb kernel-identify.
‣ "bootctl kernel-identify /var/tmp/mkosi-workspacev60pcgez/root/usr/lib/modules/6.6.14-200.fc39.s390x/vmlinuz" returned non-zero exit code 1.
root@zenv:~/mkosi-main# uname -a && cat /etc/os-release
Linux zenv 5.15.0-89-generic #99-Ubuntu SMP Mon Oct 30 20:42:29 UTC 2023 s390x s390x s390x GNU/Linux
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

**Comment by lysliu** [comment]
> Contributor

same error after `apt install qemu-user`

**Comment by DaanDeMeyer** [comment]
@lysliu Try installing qemu-user-binfmt as well

File: mkosi/qemu.py [hunk 1]
```diff
@@ -139,8 +140,13 @@ class KernelType(StrEnum):
...
     @classmethod
     def identify(cls, config: Config, path: Path) -> "KernelType":
+        if systemd_tool_version(config, "bootctl") < 253:
+            logging.warning("bootctl doesn't know kernel-identify verb, assuming 'unknown' kernel type")
+            return KernelType.unknown
+
         type = run(["bootctl", "kernel-identify", path],
-                   stdout=subprocess.PIPE, sandbox=config.sandbox(options=["--ro-bind", path, path])).stdout.strip()
+                    stdout=subprocess.PIPE,
+                    sandbox=config.sandbox(options=["--ro-bind", path, path])).stdout.strip()
```
Comment (behrmann) [inline mkosi/qemu.py:149]: Always a pity that suggestions don't work when including an unchanged line.
```
type = run(
    ["bootctl", "kernel-identify", path],
    stdout=subprocess.PIPE,
    sandbox=config.sandbox(options=["--ro-bind", path, path])
).stdout.strip()

**Review by behrmann — APPROVED** [review]

**Pull request merged**